core/include/scipp/core/element/arithmetic.h

```diff
@@ -14,13 +14,32 @@ namespace scipp::core::element {
 
 constexpr auto add_inplace_types =
     arg_list<double, float, int64_t, int32_t, Eigen::Vector3d,
+             std::tuple<scipp::core::time_point, int64_t>,
+             std::tuple<scipp::core::time_point, int32_t>,
              std::tuple<double, float>, std::tuple<int64_t, int32_t>,
              std::tuple<int64_t, bool>>;
 
-constexpr auto plus_equals =
-    overloaded{add_inplace_types, [](auto &&a, const auto &b) { a += b; }};
-constexpr auto minus_equals =
-    overloaded{add_inplace_types, [](auto &&a, const auto &b) { a -= b; }};
+constexpr auto plus_equals = overloaded{
+    add_inplace_types,
+    [](auto &&a, const auto &b) { a += b; },
+    [](scipp::core::time_point &a, const int64_t &b) {
```

Comment:
can you drop the following lambda if you change this to be `[](scipp::core::time_point &a, const auto &b) {}`? If so, could do the same for `minus_equals` below and anywhere else.

Reply:
good suggestion! thanks.